frontend/src/pages/projects/screens/spawner/environmentVariables/GenericKeyValuePairField.tsx

```diff
@@ -33,13 +33,14 @@ const GenericKeyValuePairField: React.FC<GenericKeyValuePairFieldProps> = ({
       <StackItem>
         <Stack hasGutter>
           {values.map(({ key, value }, i) => (
-            <StackItem key={i}>
+            <StackItem key={i} data-testid={`key-value-pair ${i}`}>
```

Comment:
Unnecessary to add the index to the test ID as you can use cypress to get the nth matching item.
https://docs.cypress.io/api/commands/eq